pygmt/helpers/utils.py

```diff
@@ -91,6 +92,84 @@ def data_kind(data, x=None, y=None, z=None, required_z=False):
     return kind
 
 
+def non_ascii_to_octal(argstr):
+    r"""
+    Translate non-ASCII characters to their corresponding octal codes.
+
+    Currently, only the ISOLatin1+ character set is supported.
+
+    References:
+
+    - https://docs.generic-mapping-tools.org/latest/cookbook/octal-codes.html
+    - https://www.ascii-code.com/ISO-8859-1
+    - https://www.adobe.com/jp/print/postscript/pdfs/PLRM.pdf
+
+    Parameters
+    ----------
+    argstr : str
+        The string to be translated.
+
+    Returns
+    -------
+    str
+        The translated string.
+
+    Examples
+    --------
+    >>> non_ascii_to_octal("•‰“”±°ÿ")
+    '\\31\\214\\216\\217\\261\\260\\377'
+    >>> non_ascii_to_octal("αζ∆Ω∑∏∇")
+    '@~\\141@~@~\\172@~@~\\104@~Ω@~\\345@~@~\\325@~@~\\321@~
+    >>> non_ascii_to_octal("✁❞❡➾")
+    '@%34%\\41@%%@%34%\\176@%%@%34%\\241@%%@%34%\\376@%%'
+    >>> non_ascii_to_octal("ABC ±120° DEF α ♥")
+    'ABC \\261120\\260 DEF @~\\141@~ @%34%\\252@%%'
+    """
+    # pylint: disable=line-too-long
+
+    # Dictionary mapping non-ASCII characters to octal codes
+    mapping = {}
+
+    # Symbol charset: \041-\176 and \240-\376
+    mapping.update(
+        {
+            c: "@~\\" + format(i, "o") + "@~"
+            for c, i in zip(
+                "!∀#∃%&∋()∗+,−./0123456789:;<=>?≅ΑΒΧ∆ΕΦΓΗΙθΚΛΜΝΟΠΘΡΣΤΥςΩΞΨΖ[∴]⊥_αβχδεφγηιφκλμνοπθρστυϖωξψζ{|}∼€Υ′≤⁄∞ƒ♣♦♥♠↔←↑→↓°±″≥×∝∂•÷≠≡≈…↵אIR℘⊗⊕∅∩∪⊃⊇⊄⊂⊆∈∉∠∇∏√⋅¬∧∨⇔⇐⇑⇒⇓◊〈∑ 〉∫⌠⌡",
+                [*range(33, 127), *range(160, 255)],
+            )
+        }
+    )
+
+    # ZapfDingbats charset: \041-\176 and \240-\376
+    mapping.update(
+        {
+            c: "@%34%\\" + format(i, "o") + "@%%"
+            for c, i in zip(
+                "✁✂✃✄☎✆✇✈✉☛☞✌✍✎✏✐✑✒✓✔✕✖✗✘✙✚✛✜✝✞✟✠✡✢✣✤✥✦✧★✩✪✫✬✭✮✯✰✱✲✳✴✵✶✷✸✹✺✻✼✽✾✿❀❁❂❃❄❅❆❇❈❉❊❋●❍■❏❐❑❒▲▼◆❖◗❘❙❚❛❜❝❞❡❢❣❤❥❦❧♣♦♥♠①②③④⑤⑥⑦⑧⑨⑩❶❷❸❹❺❻❼❽❾❿➀➁➂➃➄➅➆➇➈➉➊➋➌➍➎➏➐➑➒➓➔→↔↕➘➙➚➛➜➝➞➟➠➡➢➣➤➥➦➧➨➩➪➫➬➭➮➯ ➱➲➳➴➵➶➷➸➹➺➻➼➽➾",
+                [*range(33, 127), *range(161, 255)],
+            )
+        }
+    )
+
+    # ISOLatin1+ charset: \031-\037 and \177-\237
+    mapping.update(
+        {
+            c: "\\" + format(i, "o")
+            for c, i in zip(
+                "•…™—–ﬁžšŒ†‡Ł⁄‹Š›œŸŽł‰„“”ı`´ˆ˜¯˘˙¨‚˚¸'˝˛ˇ",
+                [*range(25, 32), *range(127, 160)],
+            )
+        }
+    )
+    # ISOLatin1+ charset: \240-\377
+    mapping.update({chr(i): "\\" + format(i, "o") for i in range(160, 256)})
```

Comment:
Actually, Python supports many different encodings, thus it's possible to implement this feature without manually maintaining the big dictionary, just like what I already do at line 166 for ISOLatin1+ characters \240 to \377.

Reply:
**Update:**

Can be done using:
```python
for i in range(160, 256):
    print(i, chr(i), chr(i).encode("iso-8859-1").decode("iso-8859-5"))
```

Reply:
Here is the improved version:
```
for code in [*range(0o040, 0o200), *range(0o240, 0o400)]:
    char = codecs.decode(bytes([code]), f"iso8859-{i}", errors="replace")
```